geode/dispatcher.py

```diff
@@ -254,7 +262,10 @@ def __init__(self, config=None):
         else:
             merged_df = pd.concat([estimate_df, res_df], sort=False)
 
-        merged_df = merged_df[~merged_df.index.duplicated(keep='last')].sort_index()
+        merged_df = merged_df[~merged_df.index.duplicated(keep='last')].reindex(index=idx.index, copy=False)
```

Comment:
If we're doing `~merged_df.index.duplicated(keep='last')`, is this just dropping the last?

Reply:
```duplicated()``` marks all duplicates True
```duplicated(keep='last')``` marks all except last True
with ~ the duplicate rows become false (except last occurrence)

Reply:
Cute.